src/java/com/zimbra/bp/BulkDownloadServlet.java

```diff
@@ -21,89 +21,89 @@
 
 public class BulkDownloadServlet extends ExtensionHttpHandler {
     public static final String HANDLER_NAME = "bulkdownload";
-    public static final String ACTION_GETSR = "getSR"; // get search results
-    private static final String ACTION_GETBP_FILE = "getBulkFile"; // get search results
     private static final String fileFormat = "fileFormat";
     private static final String fileID = "fileID";
 
     @Override
     public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
         try {
+            ZimbraLog.extensions.debug("Received a download request for a bulk provision file or an import results");
+
+            String pFileId = req.getParameter(fileID);
+            String pFileFormat = req.getParameter(fileFormat);
+            String bulkFileName = null;
+            String clientFileName = null;
+
+            try {
+                Double.parseDouble(pFileId);
+            } catch (Exception e) {
+                sendError(resp, HttpServletResponse.SC_BAD_REQUEST, "Bad parameter");
+                return;
+            }
+            boolean isImportResultsDownload =
+                pFileFormat.equalsIgnoreCase(ZimbraBulkProvisionExt.FILE_FORMAT_BULK_IMPORT_ERRORS) ||
+                pFileFormat.equalsIgnoreCase(ZimbraBulkProvisionExt.FILE_FORMAT_BULK_IMPORT_REPORT);
+
             // check the auth token
             AuthToken authToken = ZimbraServlet.getAdminAuthTokenFromCookie(req, resp);
-            if (authToken == null || !authToken.isAdmin()) {
+            if (authToken == null ||
+                isImportResultsDownload && !AuthToken.isAnyAdmin(authToken) ||
+                !isImportResultsDownload && !authToken.isAdmin()) {
                 sendError(resp, HttpServletResponse.SC_FORBIDDEN, "Auth failed");
+                return;
             }
 
-            String action = req.getParameter("action");
-            if (action != null && action.length() > 0) {
-                ZimbraLog.extensions.debug("Received a file download request with action = %s", action);
-            } else {
-                return;
+            // check rights
+            if (isImportResultsDownload && !authToken.isAdmin()) {
```

Comment:
I think this block can be removed. If the admin was able to generate the file, the same admin should be able to download it.

Reply:
Removed the block.